CandidateExercise/ConfigurationExercise/Config.cs

```diff
@@ -88,22 +88,60 @@ public void Shutdown()
 
         public ConfigValue GetValue(string key)
         {
-            throw new NotImplementedException();
+            // Creating a new List and adding the key in the list
+            IList<string> keyList = new List<string>(); keyList.Add(key);
+
+            if(_persistSource != null) {
```

Comment:
The initialization of your persistSource should always be performed (it should never be null), so your individual methods shouldn't need to check for its existence. I would recommend a try/catch instead of null check, that way you are protected from more than just a null object.